src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -689,6 +708,163 @@ MarkNodeHasMetadata(char *nodeName, int32 nodePort, bool hasMetadata)
 
 
 /*
+ * SequenceDDLCommandsForTable returns a list of commands which create sequences (and
+ * their schemas) to run on workers before creating the relation. The sequence creation
+ * commands are wrapped with a `worker_apply_sequence_command` call, which sets the
+ * sequence space uniquely for each worker.
+ */
+List *
+SequenceDDLCommandsForTable(Oid relationId)
+{
+	List *sequenceDDLList = NIL;
+	List *ownedSequences = getOwnedSequences(relationId);
+	ListCell *listCell;
+	char *ownerName = TableOwner(relationId);
+	bool hasMetadataWorkers = HasMetadataWorkers();
+
+	foreach(listCell, ownedSequences)
+	{
+		Oid sequenceOid = (Oid) lfirst_oid(listCell);
+		char *sequenceDef = pg_get_sequencedef_string(sequenceOid);
+		char *escapedSequenceDef = quote_literal_cstr(sequenceDef);
+		StringInfo wrappedSequenceDef = makeStringInfo();
+		StringInfo sequenceGrantStmt = makeStringInfo();
+		Oid schemaId = InvalidOid;
+		char *createSchemaCommand = NULL;
+		char *sequenceName = get_rel_name(sequenceOid);
+
+		if (hasMetadataWorkers)
+		{
+			EnsureSupportedSequenceColumnType(sequenceOid);
+
+			/* create schema if needed */
+			schemaId = get_rel_namespace(sequenceOid);
+			createSchemaCommand = CreateSchemaDDLCommand(schemaId);
+			if (createSchemaCommand != NULL)
+			{
+				sequenceDDLList = lappend(sequenceDDLList, createSchemaCommand);
+			}
+
+			appendStringInfo(wrappedSequenceDef,
+							 WORKER_APPLY_SEQUENCE_COMMAND,
+							 escapedSequenceDef);
+
+			appendStringInfo(sequenceGrantStmt,
+							 "GRANT ALL ON SEQUENCE %s TO %s", sequenceName, ownerName);
```

Comment:
Suprised to see that `ALTER SEQUENCE .. OWNED BY` and `ALTER SEQUENCE .. OWNER TO` are different things 😄